chacha20/src/lib.rs

```diff
@@ -203,7 +229,24 @@ impl NewStreamCipher for ChaCha20 {
     }
 }
 
-impl SyncStreamCipherSeek for ChaCha20 {
+impl NewStreamCipher for ChaCha20<U12> {
+    /// Key size in bytes
+    type KeySize = U32;
+    /// Nonce size in bytes
+    type NonceSize = U12;
+
+    fn new(key: &GenericArray<u8, Self::KeySize>, iv: &GenericArray<u8, Self::NonceSize>) -> Self {
+        let mut out = ChaCha20 {
+            state: ChaChaState::new(key, iv),
+        };
+
+        out.gen_block();
+
+        out
+    }
+}
+
+impl<N: ArrayLength<u8>> SyncStreamCipherSeek for ChaCha20<N> {
```

Comment:
Looking at this again, I think I still have to split this up for U8 and U12 because of encoding part of the nonce in the block index.

Reply:
It seems like I misunderstood how the `block_idx` and `seek` interact. I've added some more tests to convince myself that the implementation is correct.